webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala

```diff
@@ -0,0 +1,1424 @@
+            _ <- Future(if (userIri.isEmpty) throw BadRequestException("User IRI cannot be empty."))
+            _ = if (projectIri.isEmpty) throw BadRequestException("Project IRI cannot be empty")
+
+            // check if the requesting user is allowed to perform updates
+            _ = if (!requestingUser.permissions.isProjectAdmin(projectIri) && !requestingUser.permissions.isSystemAdmin) {
+                // not a project or system admin
+                // log.debug("project admin: {}, system admin: {}", userProfileV1.permissionData.isProjectAdmin(projectIri), userProfileV1.permissionData.isSystemAdmin)
+                throw ForbiddenException("User's project membership can only be changed by a project or system administrator")
+            }
+
+            // check if user exists
+            userExists <- userExists(userIri)
+            _ = if (!userExists) throw NotFoundException(s"The user $userIri does not exist.")
+
+            // check if project exists
+            projectExists <- projectExists(projectIri)
+            _ = if (!projectExists) throw NotFoundException(s"The project $projectIri does not exist.")
+
+            // get users current project membership list
+            currentProjectMemberships <- userProjectMembershipsGetRequestADM(
+                userIri = userIri,
+                requestingUser = KnoraSystemInstances.Users.SystemUser,
+                apiRequestID = apiRequestID
+            )
+
+            currentProjectMembershipIris: Seq[IRI] = currentProjectMemberships.projects.map(_.id)
+
+            // check if user is already member and if not then append to list
+            updatedProjectMembershipIris = if (!currentProjectMembershipIris.contains(projectIri)) {
+                currentProjectMembershipIris :+ projectIri
+            } else {
+                throw BadRequestException(s"User $userIri is already member of project $projectIri.")
+            }
+
+            // create the update request
+            userUpdatePayload = UserUpdatePayloadADM(projects = Some(updatedProjectMembershipIris))
+
+            result <- updateUserADM(userIri, userUpdatePayload, requestingUser, apiRequestID)
+
+        } yield result
+
+
+        for {
+            // run the task with an IRI lock
+            taskResult <- IriLocker.runWithIriLock(
+                apiRequestID,
+                userIri,
+                () => userProjectMembershipAddRequestTask(userIri, projectIri, requestingUser, apiRequestID)
+            )
+        } yield taskResult
+
+    }
+
+    /**
+      * Removes a user from a project.
+      *
+      * @param userIri       the user's IRI.
+      * @param projectIri    the project's IRI.
+      * @param requestingUser the requesting user.
+      * @param apiRequestID  the unique api request ID.
+      * @return
+      */
+    def userProjectMembershipRemoveRequestV1(userIri: IRI, projectIri: IRI, requestingUser: UserADM, apiRequestID: UUID): Future[UserOperationResponseADM] = {
```

Comment:
Why isn't this method called `userProjectMembershipRemoveRequestADM`?

Reply:
forgot to rename it.

Reply:
done